temporal-kotlin/src/main/kotlin/io/temporal/worker/WorkerExt.kt

```diff
@@ -66,7 +66,12 @@ inline fun <reified T : Any> Worker.registerWorkflowImplementationType(
  * @param factory factory that when called creates a new instance of the workflow implementation
  * object.
  * @see Worker.addWorkflowImplementationFactory
+ * @deprecated See deprecation notes on [Worker.addWorkflowImplementationFactory]
  */
+@Deprecated(
+  "Use Worker.registerWorkflowImplementationFactory instead",
+  ReplaceWith("Worker.registerWorkflowImplementationTypes(factory, options)")
+)
```

Comment:
The `ReplaceWith` annotation could probably be adjusted
* as far as I understand it should reference `registerWorkflowImplementationFactory` instead of `registerWorkflowImplementationTypes`
* the `Worker` instance acts as `this` for the extension method, so the replacement can refer to it like that. Otheriwse, IDEA would insert a literal `Worker` as a replacement
![CleanShot 2022-09-08 at 14 18 18](https://user-images.githubusercontent.com/61568/189109182-0485d354-1486-432c-84ed-4ecf68aa6731.gif)
```suggestion
@Deprecated(
  "Use Worker.registerWorkflowImplementationFactory instead",
  ReplaceWith("this.registerWorkflowImplementationFactory(factory, options)")
)
```